python/cuml/dask/ensemble/randomforestregressor.py

```diff
@@ -242,7 +271,14 @@ def _func_build_rf(
         )
 
     @staticmethod
-    def _fit(model, X_df, y_df, r):
+    def _fit(model, X_df_list, y_df_list, r):
+        assert len(X_df_list) == len(y_df_list)
```

Comment:
I think raising a `ValueError` here rather than assert would be better, and would allow us to use custom exceptions in the future